src/zmx2batoid/zmx_parsers.py

```diff
@@ -392,6 +169,375 @@ def extract_pupils(self):
                     except ValueError:
                         print("Warning: Could not convert Exit Pupil Diameter to float.")
 
+    def extract_fields(self):
+        """Extract and return field coordinates from the system data."""
+        fields = []
+
+        with open(self.file_path, 'r') as file:
+            lines = file.readlines()
+
+            field_i = 1
```

Comment:
I think a comment or two would really help someone understand this bit  (and below).

Reply:
Looking below I see at least 4 other places you invoke this same type of logic handling. Presumably it's all the same functionality? Maybe write an explanation in one and then just say, "see comments in XYZ" in the others?
Open to other ideas.